core/tests/test_transform.h

```diff
@@ -0,0 +1,265 @@
+    }
+
+    // set position
+    {
+        Transform* t = transform_make_default();
+        TEST_ASSERT(t != NULL);
+        transform_set_position(t, 1.0f, 2.0f, 3.0f);
+        const float3* pos1 = transform_get_position(t);
+        const float3 expected = { 1.0f, 2.0f, 3.0f };
+        TEST_CHECK(float3_isEqual(pos1, &expected, EPSILON_ZERO));
+        transform_release(t);
+    }
+
+    // set position (with a float3)
+    {
+        Transform* t = transform_make_default();
+        TEST_ASSERT(t != NULL);
+        const float3* pos1 = transform_get_position(t);
+        float3* pos2 = float3_new(4.0f, 5.0f, 6.0f);
+        const float3 expected = { 4.0f, 5.0f, 6.0f };
+        transform_set_position_vec(t, pos2);
+        TEST_CHECK(float3_isEqual(pos1, &expected, EPSILON_ZERO));
+
+        float3_free(pos2);
+        transform_release(t);
+    }
+
+    // set child local position
+    {
+        Transform* p = transform_make_default();
+        Transform* c = transform_make_default();
+        TEST_ASSERT(p != NULL && c != NULL);
+
+        transform_set_parent(c, p, true);
+
+        transform_set_position(p, 2.0f, 4.0f, 6.0f);
+        transform_set_local_position(c, -1.0f, -2.0f, -3.0f);
+
+        const float3* result = transform_get_position(c);
+        const float3 expected = { 1.0f, 2.0f, 3.0f };
+        TEST_CHECK(float3_isEqual(result, &expected, EPSILON_ZERO));
+
+        transform_release(p);
+        transform_release(c);
+    }
+}
+
+void test_transform_child(void) {
+    // check local position / rotation
+    {
+        Transform* t = transform_make_default();
+        Transform* child = transform_make_default();
+        TEST_ASSERT(t != NULL && child != NULL);
+        transform_set_parent(child, t, true);
+        TEST_CHECK(transform_get_parent(child) == t);
+
+        transform_set_position(t, 2.0f, 4.0f, 6.0f);
+        transform_set_local_position(child, 1.0f, 2.0f, 3.0f);
+        const float3* pos1 = transform_get_local_position(child);
+        const float3 expected1 = { 1.0f, 2.0f, 3.0f };
+        TEST_CHECK(float3_isEqual(pos1, &expected1, EPSILON_ZERO));
+
+        const float3* pos2 = transform_get_position(child);
+        const float3 expected2 = { 3.0f, 6.0f, 9.0f };
+        TEST_CHECK(float3_isEqual(pos2, &expected2, EPSILON_ZERO));
+
+        Quaternion* rot1 = quaternion_new(PI_F * 0.5f, 0.0f, 0.0f, 0.0f, false);
+        transform_set_rotation(t, rot1);
+        transform_set_local_rotation(child, rot1);
+        const Quaternion* rot2 = transform_get_rotation(child);
+        Quaternion* expected3 = quaternion_new(0.0f, 0.0f, 0.0f, -PI_F, false);
+        TEST_CHECK(quaternion_is_equal(rot2, expected3, EPSILON_QUATERNION_ERROR));
+        quaternion_free(rot1);
+        quaternion_free(expected3);
+
+        transform_set_rotation_euler(t, 0.0f, PI_F * 0.25f, 0.0f);
+        transform_set_local_rotation_euler(child, 0.0f, PI_F * 0.125f, 0.0f);
+        float3* rot4 = float3_new_zero();
```

Comment:
Same here. No need for a malloc here. 

`float3 rot4 = float3_zero();` should be enough